airbyte-webapp/src/hooks/services/FormChangeTracker/FormChangeTrackerService.tsx

```diff
@@ -0,0 +1,38 @@
+import type { Transition } from "history";
+
+import React, { useCallback, useMemo } from "react";
+
+import { useBlocker } from "hooks/router/useBlocker";
+
+import { useConfirmationModalService } from "../ConfirmationModal";
+import { useChangedFormsById } from "./hooks";
+
+export const FormChangeTrackerService: React.FC = ({ children }) => {
+  const [changedFormsById, setChangedFormsById] = useChangedFormsById();
+  const { openConfirmationModal, closeConfirmationModal } = useConfirmationModalService();
+
+  const blocker = useCallback(
+    (tx: Transition) => {
+      openConfirmationModal({
+        title: "form.discardChanges",
+        text: "form.discardChangesConfirmation",
+        submitButtonText: "form.discardChanges",
+        onSubmit: async () => {
```

Comment:
I don't know if this needs to be async? Doesn't hurt anything as far as I can tell.